src/cmake_build_extension/build_extension.py

```diff
@@ -161,6 +161,10 @@ def build_extension(self, ext: CMakeExtension) -> None:
         # CMake build arguments
         build_args = ["--config", ext.cmake_build_type]
 
+        # If CMAKE_BUILD_PARALLEL_LEVEL is not set, apply the corresponding user settings.
+        if "CMAKE_BUILD_PARALLEL_LEVEL" not in os.environ:
```

Comment:
Reminder: this works only with [CMake >= 3.12](https://cmake.org/cmake/help/latest/envvar/CMAKE_BUILD_PARALLEL_LEVEL.html), if we end up using this env variable, we need to document somewhere that at least 3.12 is required.